nixos/tests/cloud-init.nix

```diff
@@ -49,19 +49,17 @@ let
               gateway: '12.34.56.9'
           - type: nameserver
             address:
-            - '8.8.8.8'
+            - '6.7.8.9'
```

Comment:
Why this change? It's not clear to me what the difference is

Reply:
Because systemd-resolved automatically adds cloudflare DNS (1.1.1.1) and google DNS (8.8.8.8) as fallback. I wanted to be sure that it is picking up the nameserver from cloud-init instead of using the fallback values.

Reply:
Might be good to document that

Reply:
Sure, I'll add a comment in the next PR involving cloud-init